platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/merkle/disk/blockstream/StateChangesObserverSingleton.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.merkle.disk.blockstream;
+
+import com.hedera.node.config.ConfigProvider;
+import com.hedera.node.config.data.BlockRecordStreamConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Singleton instance of the StateChangesObserver.
+ */
+public class StateChangesObserverSingleton {
```

Comment:
We are using this approach so that if we are on `recordFileVersion` below `7`, we would want to have a No-Op observer of the changes. We want the observer to track the state changes from version `7` and ongoing